### PR TITLE
Delete block

### PR DESCRIPTION
**Pull request by geenen124**
Add a delete button which becomes active when there is a selected camera shot. When pressed the shot is deleted from the view & from memory.

### CODE REVIEW

File: src/main/java/gui/CameraShotBlock.java [hunk 1]
```diff
@@ -90,4 +90,8 @@ public ShotblockUpdatedEvent getShotBlockUpdatedEvent() {
     public CameraShot getShot() {
         return (CameraShot) super.getShot();
     }
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/CameraShotBlock.java:93]: missing javadoc

**Comment by bart-de-jonge** [comment]
Looks good, please take a look at my single comment

File: src/main/java/gui/CameraShotBlock.java (hunk 1 above)
Comment (geenen124) [inline src/main/java/gui/CameraShotBlock.java:93]: Noted

**Comment by Markvanderuit** [comment]
PR looks good, I don't really have anything to say about it except for what Bart already said.

**Comment by martijnjanssen** [comment]
I have seen a lot of null-checks on the getter of activeShotBlock, and a lot don't have a null-check. Would it be a good idea to change the return type of the getter to an `Optional<ShotBlock>`? This removes all the weird checks and lets everyone know that there is not always a value present.

**Comment by martijn9612** [comment]
Why don't you change the `enableButton` and `disableButton` to a toggle action? Would remove 1 method and it is equally clear on what it does.

**Comment by bart-de-jonge** [comment]
To me that seems like a good plan, but maybe make that an issue since it might be some work.

**Comment by bart-de-jonge** [comment]
I don't think toggle is a good idea, if you call enable or disable twice in a row things might get messed up.

**Comment by martijnjanssen** [comment]
I deleted my comment. I saw that it does not comply with the way the code should work.

**Comment by martijnjanssen** [comment]
Creating an optional is just changing the return type to an `Optional<T>`. Like this:
image

**Pull request merged**